File: code/lab2.c
#include <wiringPi.h>
#include <iostream>

#define RED_PIN 19
#define GREEN_PIN 20

int main()
{
    wiringPiSetupGpio();
    pinMode(RED_PIN, OUTPUT);
    pinMode(GREEN_PIN, OUTPUT);

    while (true)
    {
        digitalWrite(RED_PIN, HIGH);
        digitalWrite(GREEN_PIN, LOW);
        std::cout << "Red LED is ON" << std::endl;
        delay(1000);

        digitalWrite(RED_PIN, LOW);
        digitalWrite(GREEN_PIN, HIGH);
        std::cout << "Green LED is ON" << std::endl;
        delay(1000);
    }

    return 0;
}

File: code/lab2.py
import RPi.GPIO as GPIO
import time

RED_PIN = 19  # 红色LED
GREEN_PIN = 20  # 绿色LED

# 设置GPIO模式
GPIO.setmode(GPIO.BCM)
GPIO.setup(RED_PIN, GPIO.OUT)
GPIO.setup(GREEN_PIN, GPIO.OUT)

try:
    while True:
        # 打开红色LED
        GPIO.output(RED_PIN, GPIO.HIGH)
        GPIO.output(GREEN_PIN, GPIO.LOW)
        print("红色LED已打开")
        time.sleep(1)  # 等待1秒钟

        # 打开绿色LED
        GPIO.output(RED_PIN, GPIO.LOW)
        GPIO.output(GREEN_PIN, GPIO.HIGH)
        print("绿色LED已打开")
        time.sleep(1)  # 等待1秒

except KeyboardInterrupt:
    print("程序被终止")

finally:
    # 退出前清理GPIO设置
    GPIO.cleanup()
